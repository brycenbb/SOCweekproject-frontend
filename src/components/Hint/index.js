import React from 'react';

function Hint({ index, text }) {
  return (
    <li className={getClass(index)} id={`checklist${index}`} key={index}>
      <label className={'hint-list-item'} htmlFor={index}>
        {text.hint}
      </label>
      <input
        className="hint-Checkbox"
        name={index}
        type="checkbox"
        id={`checkbox${index}`}
        onClick={() => nextHint(index)}
      ></input>
    </li>
  );
}

//Helper Function
function nextHint(currIndex) {
  if (
    currIndex ===
    document.querySelector('.list-of-hints').childNodes.length - 1
  ) {
    return;
  }
  let checkBox = document.getElementById(`checkbox${currIndex}`);
  let list = document.getElementById(`checklist${currIndex + 1}`);
  if (checkBox.checked === true) {
    list.classList.remove('hidden');
  } else {
    list.classList.add('hidden');
  }
}

//Helper Function
function getClass(index) {
  if (index === 0) {
    return;
  }
  return 'hidden';
}
export default Hint;
