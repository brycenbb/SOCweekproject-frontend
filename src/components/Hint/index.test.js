import Hint from './index.js';
import { useState } from 'react';
import { test, expect } from '@jest/globals';

const STATUS = {
  CLICKED: 'clicked',
  UNCLICKED: 'unclicked',
};

test('checkbox is clicked', () => {
  const checkBox = { Hint };
  let tree = checkBox.toJSON;
  expect(tree).toMatchSnapshot();
});
