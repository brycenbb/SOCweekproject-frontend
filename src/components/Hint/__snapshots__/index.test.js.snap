// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checkbox is clicked 1`] = `undefined`;
